### ISSUE TITLE
The current N4 bias correction binary is not deterministic across platforms

### ISSUE THREAD

**Issue by ledigchr** [ledigchr/MALPEM#8]
This is an important issue reported with suggested fix here:
https://github.com/MIRTK/MALPEM/issues/2

**Comment by ledigchr** [comment]
Some experiments around that seem to confirm what is also discussed in this [thread](https://sourceforge.net/p/advants/discussion/840261/thread/354df4c0/?limit=25#116c) by the N4 developers.

* The N4 binary that is shipped with MALPEM yields slightly different results if it runs with a varying number of Threads - seems to be a problem with ITK. 
* The number of ITK Threads defaults to the number of CPUs on the system. So on the same system N4 (and as consequence MALPEM) should produce the same results.
* The number of Threads can be specified with the environment variable `ITK_GLOBAL_DEFAULT_NUMBER_OF_THREADS`. So if this is set explicitly, N4 should be reproducible across systems with varying number of CPUs. 
* I've tested this on AWS instances and reproduced a default run on `m5a.8xlarge` (32 CPUs) on a `t2.xlarge` (4 CPUs, but with `ITK_GLOBAL_DEFAULT_NUMBER_OF_THREADS=32`). 

My suggested fix for the MALPEM package is to update `malpem-proot` and explicitly set `ITK_GLOBAL_DEFAULT_NUMBER_OF_THREADS`. E.g.

```
'XAUTHORITY=/tmp/.mdmCHXAXX'  \
"${PROOT}" \
```
-->
```
'XAUTHORITY=/tmp/.mdmCHXAXX'  \
'ITK_GLOBAL_DEFAULT_NUMBER_OF_THREADS=32' \
"${PROOT}" \
```

**Comment by ledigchr** [comment]
Fixed in [malpem-v1.3](https://github.com/ledigchr/MALPEM/releases/tag/v1.3) by setting `'ITK_GLOBAL_DEFAULT_NUMBER_OF_THREADS=8'` in `malpem-proot`
